PHPstan errors in az_course

## Problem/Motivation
After we merged #1594,  PHPstan errors are found on new PRs.

```
 ------ --------------------------------------------------------------- 
  Line   modules/custom/az_course/src/CourseMigrateBatchExecutable.php  
 ------ --------------------------------------------------------------- 
  25     Call to an undefined method                                    
         Drupal\migrate\Plugin\MigrationInterface::set().               
 ------ --------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------- 
  Line   modules/custom/az_course/src/CourseSearch.php                         
 ------ ---------------------------------------------------------------------- 
  34     PHPDoc tag @param for parameter $http_client with type                
         Drupal\az_course\GuzzleHttp\ClientInterface is not subtype of native  
         type GuzzleHttp\ClientInterface.                                      
  34     Parameter $http_client of method                                      
         Drupal\az_course\CourseSearch::__construct() has invalid type         
         Drupal\az_course\GuzzleHttp\ClientInterface.                          
  84     Call to an undefined method GuzzleHttp\ClientInterface::get().        
 ------ ---------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------- 
  Line   modules/custom/az_course/src/Form/CourseImportForm.php          
 ------ ---------------------------------------------------------------- 
  199    Variable $migration in empty() always exists and is not falsy.  
  206    Variable $course_urls might not be defined.                     
 ------ ---------------------------------------------------------------- 

 ------ ----------------------------------------------------------------------- 
  Line   modules/custom/az_course/src/Plugin/migrate_plus/data_fetcher/RetryHt  
         tp.php                                                                 
 ------ ----------------------------------------------------------------------- 
  51     Variable $response in empty() always exists and is not falsy.          
 ------ ----------------------------------------------------------------------- 

 [ERROR] Found 7 errors 
```

## Proposed resolution
Address PHPstan errors.

--- a/modules/custom/az_course/src/CourseMigrateBatchExecutable.php
+++ b/modules/custom/az_course/src/CourseMigrateBatchExecutable.php
@@ -21,21 +21,16 @@ class CourseMigrateBatchExecutable extends MigrateBatchExecutable {
         $migration->getIdMap()->prepareUpdate();
       }
 
-      if (!empty($options['force'])) {
-        $migration->set('requirements', []);
-      }
-      else {
-        $dependencies = $migration->getMigrationDependencies();
-        if (!empty($dependencies['required'])) {
-          $required_migrations = $this->migrationPluginManager->createInstances($dependencies['required']);
-          // For dependent migrations will need to be migrate all items.
-          $operations = array_merge($operations, $this->batchOperations($required_migrations, $operation, [
-            'limit' => 0,
-            'update' => $options['update'],
-            'force' => $options['force'],
-            'sync' => $options['sync'],
-          ]));
-        }
+      $dependencies = $migration->getMigrationDependencies();
+      if (!empty($dependencies['required'])) {
+        $required_migrations = $this->migrationPluginManager->createInstances($dependencies['required']);
+        // For dependent migrations will need to be migrate all items.
+        $operations = array_merge($operations, $this->batchOperations($required_migrations, $operation, [
+          'limit' => 0,
+          'update' => $options['update'],
+          'force' => $options['force'],
+          'sync' => $options['sync'],
+        ]));
       }
 
       // Take expected source urls and run them one per batch. This is the

--- a/modules/custom/az_course/src/CourseSearch.php
+++ b/modules/custom/az_course/src/CourseSearch.php
@@ -28,7 +28,7 @@ class CourseSearch {
   /**
    * Constructs a CourseSearch service.
    *
-   * @param GuzzleHttp\ClientInterface $http_client
+   * @param \GuzzleHttp\ClientInterface $http_client
    *   A guzzle client for making http requests.
    */
   public function __construct(ClientInterface $http_client) {
@@ -81,10 +81,7 @@ class CourseSearch {
     $queryUrl = Url::fromUri('https://uacourses-api.uaccess.arizona.edu/search', ['query' => ['search_text' => $search]]);
     $queryUrl = $queryUrl->toString();
     try {
-      $response = $this->httpClient->get($queryUrl);
-      if (empty($response)) {
-        \Drupal::logger('az_course')->error("No response.");
-      }
+      $response = $this->httpClient->request('GET', $queryUrl);
       $body = $response->getBody();
       $xml = new \XMLReader();
       if ($xml->xml($body)) {

--- a/modules/custom/az_course/src/Form/CourseImportForm.php
+++ b/modules/custom/az_course/src/Form/CourseImportForm.php
@@ -174,9 +174,9 @@ class CourseImportForm extends ConfigFormBase {
       ->save();
 
     $courses = $this->config('az_course.settings')->get('courses');
+    $course_urls = [];
     if (!empty($courses)) {
       $matches = [];
-      $course_urls = [];
       // Convert courses listed into actual URLs to migrate.
       foreach ($courses as $course) {
         if (preg_match("/^[[:space:]]*([[:alpha:]]+)[[:space:]]+([[:alnum:]]+)[[:space:]]*$/", $course, $matches)) {
@@ -196,20 +196,18 @@ class CourseImportForm extends ConfigFormBase {
 
     // Pass expected urls to batched executable.
     $migration = $this->pluginManagerMigration->createInstance("az_courses");
-    if (!empty($migration)) {
-      $options = [
-        'limit' => 0,
-        'update' => 1,
-        'force' => 0,
-        'configuration' => [
-          'source' => [
-            'urls' => $course_urls,
-          ],
+    $options = [
+      'limit' => 0,
+      'update' => 1,
+      'force' => 0,
+      'configuration' => [
+        'source' => [
+          'urls' => $course_urls,
         ],
-      ];
-      $executable = new CourseMigrateBatchExecutable($migration, new MigrateMessage(), $options);
-      $executable->batchImport();
-    }
+      ],
+    ];
+    $executable = new CourseMigrateBatchExecutable($migration, new MigrateMessage(), $options);
+    $executable->batchImport();
   }
 
   /**

--- a/modules/custom/az_course/src/Plugin/migrate_plus/data_fetcher/RetryHttp.php
+++ b/modules/custom/az_course/src/Plugin/migrate_plus/data_fetcher/RetryHttp.php
@@ -48,9 +48,6 @@ class RetryHttp extends Http {
           $options = array_merge($options, $this->configuration['request_options']);
         }
         $response = $this->httpClient->get($url, $options);
-        if (empty($response)) {
-          throw new MigrateException('No response at ' . $url . '.');
-        }
         return $response;
       }
       catch (RequestException $e) {

